### PR TITLE
Fixed create_slice method to return urls with the start and end values

### LINKED ISSUE #87: CORS Request

### LINKED ISSUE #91: create_slice returns entire chromosome for 1 slice
while it should respect the input's start and end parameters..

Currently, if the requested region is smaller than the chunked size, it returns the entire chromosome

This is not going to work

E.g.

If requesting a block of 10k bps, and the chunk size is 100k bps, instead of returning data chunks of 10k bps, it returns the entire chromosome, this doesn't make sense

### LINKED ISSUE #92: Support returning binary data instead of plain text (bam instead of sam)
Currently, the `/data` endpoint defaults to returning plain text (equivalent to SAM), the endpoint should really allow for the option to return data in binary or plain text format.

Additionally, IGV browser would expect the binary (bam) formatted data, not sam.

### LINKED ISSUE #93: Invalid contig for certain bam files - reasons unknown
image

### PR DESCRIPTION
**Pull request by fcoralsasso**
This PR fixes the` create_slice` method to return URLs with the start and end values

### CODE REVIEW

File: htsget_server/operations.py [hunk 1]
```diff
@@ -340,7 +340,14 @@ def get_data(id, referenceName=None, format=None, start=None, end=None):
                 index_filename=index_file
             )
 
-        file_out = AlignmentFile(ntf.name, 'w', header=file_in.header)
+        if format == "bam":
```
Comment (jimmyhli) [inline htsget_server/operations.py]: this isn't needed if it defaults to `bam` right?

Comment (fcoralsasso) [reply htsget_server/operations.py]: That's a good point.

Comment (fcoralsasso) [reply htsget_server/operations.py]: Fixed.

**Review by jimmyhli — APPROVED** [review]

**Review by shaikh-rashid — APPROVED** [review]

**Review by jimmyhli — COMMENTED** [review]
If https://github.com/CanDIG/htsget_app/issues/87 is within the scope of this PR then we'd need to add corresponding code before this can be closed; looks good otherwise.

**Pull request merged**